contracts/src/lib.rs

```
@@ -699,7 +725,21 @@ impl PortfolioRebalancer {
             .unwrap();
         let reflector_client = ReflectorClient::new(env, &reflector_address);
 
-        let preview = portfolio::build_rebalance_preview(env, &portfolio, &reflector_client)?;
+        let triggered = stop_loss::check_stop_losses(env, portfolio_id, &portfolio, &reflector_client);
+        let has_triggered = triggered.len() > 0;
```

Clippy (1.96.0)

[warning] length comparison to zero

(warning)